src/util.cc

```diff
@@ -223,8 +223,13 @@ int WriteFileSync(v8::Isolate* isolate,
 
 int ReadFileSync(std::string* result, const char* path) {
   uv_fs_t req;
+  auto defer_req_cleanup = OnScopeLeave([&req]() {
+    uv_fs_req_cleanup(&req);
+  });
```

Comment:
An alternative approach (that may be a bit nicer here) is to use a smart-pointer type of mechanism like we use elsewhere... something like...

```cpp
struct UvFsReq {
  uv_fs_t req;
  ~UvFsReq() {
    uv_fs_req_cleanup(&req);
  }
}
```

Reply:
Sounds great, I'll do an update.

Reply:
After several tries, I found that wrapping with a naive data struct did not work well in the case. We have to access members of `uv_fs_t` and passing the pointer of `uv_fs_t` back to uv, like:

```c++
uv_fs_t req;
uv_fs_open(nullptr, &req, path, O_RDONLY, 0, nullptr);  // <- either we write with `&UvFsReq.req` or using operators to automatically convert (which might not be very straightforward since it implicit converting from data type to pointer type)
if (req.result < 0); // <- accessing member of `uv_fs_t`, either we write with UvFsReq.req.result or explicitly declare the member proxy in UvFsReq like `UvFsReq.result()`. 
```

This is too much for the intent, a `OnScopeLeave` just fit in well.

As such, I'd believe it is more readable and straightforward to use `uv_fs_t` here. And there are a lot of precedents in the code base.